File: ios-native/IronCore/Models/Battle.swift
import Foundation
import FirebaseFirestore

/// PvP battle at `battles/{battleId}`
struct Battle: Codable, Identifiable {
    @DocumentID var id: String?
    var challenger: BattlePlayer
    var opponent: BattlePlayer
    var status: String          // "pending", "active", "completed", "declined"
    var battleType: String      // "ranked", "casual"
    var winnerId: String?
    @ServerTimestamp var createdAt: Timestamp?
    @ServerTimestamp var acceptedAt: Timestamp?
    @ServerTimestamp var declinedAt: Timestamp?
    @ServerTimestamp var completedAt: Timestamp?
    @ServerTimestamp var expiresAt: Timestamp?
}

struct BattlePlayer: Codable {
    var userId: String
    var username: String
    var photo: String
    var xp: Int
}

File: android-native/app/src/main/java/com/ironcore/fit/data/model/Battle.kt
package com.ironcore.fit.data.model

import com.google.firebase.Timestamp
import com.google.firebase.firestore.ServerTimestamp

/**
 * Firestore path: /battles/{battleId}
 */
data class Battle(
    val id: String = "",
    val challenger: BattlePlayer = BattlePlayer(),
    val opponent: BattlePlayer = BattlePlayer(),
    val status: String = "pending",
    val battleType: String = "ranked",
    val winnerId: String? = null,
    @ServerTimestamp val createdAt: Timestamp? = null,
    val acceptedAt: Timestamp? = null,
    val declinedAt: Timestamp? = null,
    val completedAt: Timestamp? = null,
    val expiresAt: Timestamp? = null
)

data class BattlePlayer(
    val userId: String = "",
    val username: String = "",
    val photo: String = "",
    val xp: Long = 0
)
